base/essentials.jl

```diff
@@ -135,7 +135,13 @@ function length_checked_equal(args...)
     n
 end
 
-map(f::Function, a::Array{Any,1}) = Any[ f(a[i]) for i=1:length(a) ]
+function map(f::Function, a::Array{Any,1})
```

Comment:
The comprehension macro is not defined yet, so if I remember correctly this was an issue during bootstrap (one of the functions in the macro indirectly called this method, I don't know if that is still the case).